posthog/queries/query_date_range.py

```diff
@@ -174,21 +174,40 @@ def date_from(self) -> Tuple[str, Dict]:
 
         return date_from_query, date_from_param
 
-    def _get_timezone_aware_date_condition(self, operator: str, date_clause: str) -> str:
-        if self.should_round:
-            return f"AND toDateTime({self._table}timestamp, 'UTC') {operator} {self._timezone_date_clause(date_clause)}"
+    def _get_timezone_aware_date_condition(self, operator: Literal[">=", "<="]) -> str:
+        relevant_param = "date_from" if operator.startswith(">") else "date_to"
+        event_timestamp_expr = self._normalize_datetime(f"{self._table}timestamp", stored_in_utc=True)
+        date_expr = self._normalize_datetime(f"%({relevant_param})s")
+        if operator == ">=" and self.should_round:  # Round date_from to start of interval if `should_round` is true
```

Comment:
being unaware of the rest of the class - when is `should_round` set? Just want to make sure e.g. when selecting events for an insight that we don't round the main filter, but rather truncate later when grouping

Reply:
Honestly, I don't have a good understanding of `should_round` either. I believe this is something @EDsCODE added, perhaps we should poke his brain about this.